### DISCUSSION TITLE
dledger模式下的mq集群，重启broker主从切换会出现消息丢失

### DISCUSSION THREAD

**Discussion by LittleBoy18** [apache/rocketmq#7476] [Q&A]
1、搭建的4.9.2版本的dledger集群，发送一批消息后，重启master broker模拟主从切换，切换后发现消息有丢失
<img width="1408" alt="image" src="https://github.com/apache/rocketmq/assets/18388649/e435216f-d3d2-4bd2-afc8-502eeb17137c">

**Comment by figuewang** [comment]
主从同步是异步的，理论上是会丢消息；建议改成同步的；其次没有不丢消息的中间件，金融类建议加上本地消息表做重试

**Reply by LittleBoy18** [reply]
感谢你的回复，这个问题确认了是dledger broker在异常重启时index file可能会被删掉但是未被重建，导致根据msgId查询异常
